common/src/main/java/io/pravega/common/tracing/RequestTracker.java

```diff
@@ -38,13 +39,15 @@
 @Slf4j
 public final class RequestTracker {
 
+    @VisibleForTesting
+    static final int MAX_PARALLEL_REQUESTS = 10;
     private static final String INTER_FIELD_DELIMITER = "-";
     private static final int MAX_CACHE_SIZE = 1000000;
     private static final int EVICTION_PERIOD_MINUTES = 10;
 
     private final Object lock = new Object();
     @GuardedBy("lock")
-    private final Cache<String, List<Long>> ongoingRequests;
+    private final SimpleCache<String, List<Long>> ongoingRequests;
```

Comment:
What is the advantage of replacing google guava Cache with Simple Cache?
Google cache enables using Soft/Weak references for cache entries which enables fast reclamation of heap memory.

Reply:
My understanding is that @andreipaduroiu suggested this to unify the usage of caches across Pravega, which makes sense. Anyway, irrespective of the implementation of Guava cache, without this PR we could get OOM anyway, and I think that with this PR we will be ok in terms of heap irrespective of the cache implementation used.

Reply:
The objective of this PR is to reduce the amount of memory consumed by Request Tracker and I don;t see how this change would do that. I understand using the same library across components is a good idea but the changes in this PR need to focus on reducing the memory occupied by RequestTracker.

Reply:
Ok, we can do the switch to `SimpleCache` separately.